### PR TITLE
[RED-2026] Set a default user agent so Zendesk can identify the requests

### PR DESCRIPTION
**Pull request by fbvilela**
### Description

Sets a default user agent. 
We can now see it in our logs the requests are coming from this SDK . 

Screenshot 2023-11-14 at 3 21 44 pm

@mikerogers123 are you happy with this user agent string?

### CODE REVIEW

**Comment by mikerogers123** [comment]
@fbvilela This header is fine by me. Although, I am curious what the use case is for knowing requests are coming from this client, in particular? As I understand it, this header lets servers identify the application/OS/vendor of the requesting user. This client is just a wrapper/helper for a consuming application. Would Zendesk not be more interested in what that consuming application is? E.g. "justeat.application1"? And then Zendesk would know all requests from justeat via the `justeat.` prefix.

Also, if Zendesk did care about requests from this client, what about the case where one of our consumers chooses to overwrite this header? I believe they can using the HttpClientFactory. In this case you wouldn't know these requests were made using this library. I guess what I'm saying is: it makes sense to force consumers to use this header, if we go down this route?

Not sure if that makes sense! Happy to hear thoughts/opinions

**Comment by fbvilela** [comment]
Hi @mikerogers123 ,
Thank you for you Feedback!
User Agent value helps us to understand the usage of languages and associated tools by the developers. It also helps us to determine the level of support, planning and collaboration on changes. We are also bringing improvements to the API Usage Monitoring, so providing the signature details in the reports help customers know the traffic origins and differentiate between native HTTP library and SDK wrappers.
Currently, Zendesk is in plans reduce the availability of Offset Model in pagination in favour of Cursor Model for performance reasons. We want to ensure that customer integrations + API experience is not impacted by this change, so we are doing our best to collaborate and update the libraries that customers are using. Knowing the signature of a library through the user agent will simplify this task.
Customers should be free to change the userAgent value as they see fit. But, in our observation, majority tend to leave it to the default signature. We did not see a mechanism for users to specify a custom user Agent value in this C# library. Please advise if we missed to see it and where can one do it? We can update the readme about it

**Comment by mikerogers123** [comment]
Hi @fbvilela thanks for the context - super useful. Makes sense.

I believe it is possible for a consumer to control this header value if they were to instantiate the client via [ZendeskApiClientFactory.cs](https://github.com/justeat/ZendeskApiClient/blob/master/src/ZendeskApi.Client/ZendeskApiClientFactory.cs), but it is super unlikely seeing as the consumer would have to register a HttpClient named `zendeskApiClient` and configure that to use their custom header. So from that perspective I think your observation is effectively correct that a consumer couldn't easily change it.

So yes I am good with this change!

File: src/ZendeskApi.Client/ZendeskApiClient.cs [hunk 1]
```diff
@@ -52,6 +52,7 @@ public HttpClient CreateClient(string resource = null)
             var authorizationHeader = _options.GetAuthorizationHeader();
```
Comment (mikerogers123) [inline src/ZendeskApi.Client/ZendeskApiClient.cs:52]: Can you bump the version number please? 🙏

**Review by mikerogers123 — CHANGES_REQUESTED** [review]

File: src/ZendeskApi.Client/ZendeskApiClient.cs (hunk 1 above)
Comment (fbvilela) [reply src/ZendeskApi.Client/ZendeskApiClient.cs:52]: thank you, pushed! 🙏

**Review by mikerogers123 — APPROVED** [review]

**Pull request merged**